File: backend/app/schemas/facebook.py
"""Facebook API response schemas."""

from __future__ import annotations

from pydantic import BaseModel, ConfigDict


class FacebookAuthUrlResponse(BaseModel):
    url: str


class FacebookPageResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    page_id: str
    name: str
    username: str | None
    picture_url: str | None
    is_active: bool


class FacebookPageListResponse(BaseModel):
    items: list[FacebookPageResponse]


class CurrentFacebookPageResponse(BaseModel):
    item: FacebookPageResponse | None

File: desktop/src/types/facebook.ts
export type FacebookPage = {
  id: string;
  page_id: string;
  name: string;
  username: string | null;
  picture_url: string | null;
  is_active: boolean;
};

export type FacebookPageListResponse = {
  items: FacebookPage[];
};

export type CurrentFacebookPageResponse = {
  item: FacebookPage | null;
};

export type FacebookAuthUrlResponse = {
  url: string;
};
